### DISCUSSION TITLE
Old Domain Deprecation Notice

### DISCUSSION THREAD

**Discussion by nunopato** [nhost/nhost#1319] [General]
# Deprecation Notice

When we launched the Nhost CDN for Storage, we moved from a single domain for all services to individual domain names for each service. 

## Old path-based domain:

- https://$subdomain.nhost.run
    - /v1/auth
    - /v1/storage
    - /v1/graphql
    - /v1/functions
    - /console

## New domains:

- https://$subdomain.auth.$region.nhost.run
- https://$subdomain.storage.$region.nhost.run 
- https://$subdomain.graphql.$region.nhost.run
- https://$subdomain.functions.$region.nhost.run
- https://$subdomain.hasura.$region.nhost.run (console)

Having different domain names gives us more flexibility in terms of how we route to those endpoints.

## February 1st, 2023

The old domain will be deprecated on the 1st of February, 2023, so those of you still using it, should make sure the following requirements are met:

### `backendUrl` should be replaced with `subdomain` and `region` on the JavaScript Client. 

> The subdomain used here is only an example.

```ts
import { NhostClient } from '@nhost/nhost-js'

const nhostClient = new NhostClient({
  backendUrl: 'https://fofnewinfenfew.nhost.run',
})
```
should be replaced with:

```ts
import { NhostClient } from '@nhost/nhost-js'

const nhostClient = new NhostClient({
  subdomain: 'fofnewinfenfew',
  region: 'eu-central-1',
})
```

### Update the Redirect URL on all providers.

If you are using any OAuth provider available in the dashboard, you will have to update the configuration on the corresponding provider developer portal (or equivalent) to use the new **Redirect URL**. You can now see the updated URL on the settings page for all providers (Settings -> Sign-In Methods). Notice that the URL is specific to the auth service and respects the new domain structure. There is a confirmation dialog that you should submit once you are done. 

![image](https://user-images.githubusercontent.com/1523504/205445035-28538fa0-5afb-4e37-8fec-db7ffabadacf.png)

### Replace the environment variable `NHOST_BACKEND_URL`.

If you are using the `NHOST_BACKEND_URL` environment variable (i.e. in your event triggers), you will need to replace it with one of the new variables introduced:

- NHOST_AUTH_URL
- NHOST_STORAGE_URL
- NHOST_FUNCTIONS_URL 
- NHOST_GRAPHQL_URL

All environment variables available to your project can be found in your project's settings page, under "Environment Variables". 

As always, you can write us at support@nhost.io or on discord, if you need any help.

**Comment by OrangeNote** [comment]
I hope the following can be helpful to others ✌️ 
For some reason I had to immediately switch to the new domains, since the old redirect URLs started sending users to the Hasura console after a successful authorization, breaking the sign-in flow. Once I changed the URL for all my OAuth providers and submitted the confirmation dialogue, everything started working again. I never had issues with the old configurations for my Nhost projects until yesterday, so I suspect something went wrong with these deprecation changes, which affected two of my projects (same deployment repo, different configs).

**Reply by Alex-Ferreli** [reply]
I've the exact same problem. When i change Redirect URL on Google App, Google Login want the old Redirect URL, so login is broken. Redicrect_uri parameter on Google sign-in URL it's still the old one. Maybe the nhost package is not updated with the new structure?

UPDATE: this is the error on Google Login:
redirect_uri=https://winerqwjxpiqenrpwcbg.nhost.run/v1/auth/signin/provider/google/callback

Google Login URL is generated with old method, i don't think it's possible to change this option manually form the interface

**Reply by benorudolf** [reply]
+1 on this exact problem. 

If I add the new url to Google's Redirect urls i.e. 
```
https://<domain>.auth.<region>.nhost.run/v1/signin/provider/google/callback
```
Google Auth will error saying it expects the old url i.e. 
```
https://<domain>.nhost.run/v1/auth/signin/provider/google/callback
```

If I add the old url (or both) to Google's Redirect urls, then users get redirected to Hasura console.

**Reply by Alex-Ferreli** [reply]
I've found the problem, you must click "I have updated all Redirect URLs" to update the URLs. I think that this action change the generated URL on Google Login.

**Reply by benorudolf** [reply]
@Alex-Ferreli  Where is the action located?

**Reply by Alex-Ferreli** [reply]
@benorudolf is located in the yellow panel present in the first message of this discussion (go top to see the screenshot).

**Reply by benorudolf** [reply]
@Alex-Ferreli Of course I missed it; it's in plain sight 😂  Wonderful! Thank you!

**Reply by benorudolf** [reply]
If someone comes around here with the issue, here's the rundown: 

1. Follow [the Nhost guide](https://docs.nhost.io/authentication/sign-in-with-google) to setup your Google Auth
2. Use the provided (new) callback link from your Nhost settings when entering the callbacks inside Google Cloud Console
3. Find the yellow box in Nhost settings (just above Apple Auth) and click "I have updated all Redirect URLs" 
4. Wait 5-10mins. The Google Auth thing can take up to 5mins to refresh. 
5. (Re)Try to login with Google

**Comment by ArieJones** [comment]
So what are we using for local development then? Is the cli supposed to follow this new formula? 

I ask because we are also using an external .net core API for a portion of our project that makes calls to the graphql engine and other various things.... 

the cli looks like it still uses these old constructs... 

![image](https://user-images.githubusercontent.com/14859781/207441135-a8ea8e1f-24ef-4dbc-976c-00ba2facc03d.png)

which makes it a little harder for us because we had just created a small class in .net core to handle the normal translation ... and now we have to come up with some kind of local switch because there is a disconnect between local dev and hosted...

**Reply by nunopato** [reply]
@ArieJones the CLI is still using the path version of the domain but we might change that in the near future. Initializing the client SDK should be identical in both cases:

```
const nhostClient = new NhostClient({
  subdomain: ENV.MY_SUBDOMAIN,
  region: ENV.MY_REGION
})
```

where `MY_SUBDOMAIN` is either `localhost` for dev or a cloud instance's subdomain. `MY_REGION` is optional in development.

> which makes it a little harder for us because we had just created a small class in .net core to handle the normal translation ... and now we have to come up with some kind of local switch because there is a disconnect between local dev and hosted...

can you share more about what you are trying to do there?

**Reply by ArieJones** [reply]
We have a .net core api outside of nhost because we had a lot of pain with the functions around getting logs... debugging...etc.. 

So as part of that we have this little service written in .net core to basically interact with the various apis which is pretty straight forward since everything is based off of a single base uri .. 

with this change it seems that we would have to mimic this uri two-way scenario which kind of feels like a code smell... 

so not a big dealio really but it's kind of moving our cheese when we are in the middle of getting our app certified ..

**Reply by nunopato** [reply]
That case has to be dealt with on your .net client, yes. With `@nhost/nhost-js` that is solved for you. You only have to pass the `subdomain`. 

Seems like a .net client would be useful :)

**Comment by diegonc** [comment]
Will the [nhost-dart](https://github.com/nhost/nhost-dart/blob/main/packages/nhost_sdk/lib/nhost_sdk.dart#L57) library be updated? I don't know what's the extent of the required changes.

Where should I report this?

**Reply by mhadaily** [reply]
@diegonc It's supported already in Dart SDK in the newest DEV version. Check this out `nhost_sdk: ^4.0.0-dev.6` or [this link ](https://pub.dev/packages/nhost_sdk/versions/4.0.0-dev.6)

**Reply by diegonc** [reply]
@mhadaily  thanks for your work on this!!

I'm facing some dependency issue with the `meta` package  which is both depended on by `flutter_test` and `nhost_sdk` but at different versions.

This is the output of `flutter pub get`:
```
Running "flutter pub get" in app...                             
Because every version of flutter_test from sdk depends on meta 1.7.0 and nhost_sdk >=4.0.0-dev.0 depends on meta ^1.8.0, flutter_test from sdk is incompatible with nhost_sdk >=4.0.0-dev.0.
So, because app depends on both nhost_sdk ^4.0.0-dev.9 and flutter_test from sdk, version solving failed.
pub get failed (1; So, because app depends on both nhost_sdk ^4.0.0-dev.9 and flutter_test from sdk, version solving failed.)
Process finished with exit code 1
```

I've got the following snippet in my `pubspec.yml`:

```
environment:
  sdk: ">=2.16.2 <2.17.0"
```

2.16.2 is the latest Dart I can use due to other projects that share the SDK.

Any ideas?

**Reply by mhadaily** [reply]
can you try https://pub.dev/packages/nhost_sdk/versions/4.0.0-dev.10 and let me know if it works?

**Reply by diegonc** [reply]
Yes, it works! The dependency error is no longer produced

**Comment by pkreipke** [comment]
Late rolling back to this notice ... Feb 1 still the hard date?
Best to update SDKs in dev, backend and client apps to the latest to get proper support?

**Reply by pkreipke** [reply]
I've updated NHost in local dev and staging. Still wondering about my questions ^^^.

But also: in my React Native app, what's going to happen to users **who don't use automatic updates** if they try to run it with old URLs to `https://<domain>.nhost.run/v1/auth/signin` or `...functions` or `...graphql`?

**Reply by pkreipke** [reply]
@nunopato

**Reply by nunopato** [reply]
@pkreipke Updating the SDKs and CLI is always a good idea. If you need more time past the 1st of Feb, no worries, the old domain will be available. With that being said, I strongly encourage everyone to update as soon as possible.
